addons/account/models/account_move_line.py

```diff
@@ -1541,9 +1542,13 @@ def write(self, vals):
             if line.parent_state == 'posted' and any(self.env['account.move']._field_will_change(line, vals, field_name) for field_name in protected_fields['tax']):
                 line._check_tax_lock_date()
 
-            # Check the reconciliation.
-            if any(self.env['account.move']._field_will_change(line, vals, field_name) for field_name in protected_fields['reconciliation']):
-                line._check_reconciliation()
+            # Break the reconciliation.
+            if any(self.env['account.move']._field_will_change(line, vals, field_name) for field_name in protected_fields['reconciliation']) and (line.matched_debit_ids or line.matched_credit_ids):
+                lines_to_unreconcile += line
+
+        lines_to_unreconcile = lines_to_unreconcile._all_reconciled_lines()
+        lines_to_unreconcile.remove_move_reconcile()
+        lines_to_unreconcile.statement_line_id.action_undo_reconciliation()
```

Comment:
In case ok locked period for at least one statement in the chain, it could lead to weird behavior like the user can’t edit his invoice at all. That’s a corner case 🙄